test/unit/transloadit/test_smart_cdn.rb

```diff
@@ -0,0 +1,65 @@
+require "test_helper"
+
+describe Transloadit do
+  before do
+    @transloadit = Transloadit.new(key: "my-key", secret: "my-secret")
+    @workspace = "my-app"
+    @template = "test-smart-cdn"
+    @input = "inputs/prinsengracht.jpg"
+    @expire_at = 1732550672867
+  end
+
+  describe "#signed_smart_cdn_url" do
+    it "requires workspace" do
+      assert_raises ArgumentError, "workspace is required" do
+        @transloadit.signed_smart_cdn_url(
+          workspace: nil,
+          template: @template,
+          input: @input
+        )
+      end
+    end
+
+    it "requires template" do
+      assert_raises ArgumentError, "template is required" do
+        @transloadit.signed_smart_cdn_url(
+          workspace: @workspace,
+          template: nil,
+          input: @input
+        )
+      end
+    end
+
+    it "requires input" do
+      assert_raises ArgumentError, "input is required" do
+        @transloadit.signed_smart_cdn_url(
+          workspace: @workspace,
+          template: @template,
+          input: nil
+        )
+      end
+    end
+
+    it "uses instance credentials by default" do
+      url = @transloadit.signed_smart_cdn_url(
+        workspace: @workspace,
+        template: @template,
+        input: @input,
+        expire_at_ms: @expire_at
+      )
+      assert_match(/auth_key=my-key/, url)
+    end
+
+    it "allows overriding credentials" do
+      url = @transloadit.signed_smart_cdn_url(
```

Comment:
I wonder what the use case here is. AFAICT `signed_smart_cdn_url` is not a static method but an instance method and thus a `Transloadit` instance is always needed to call it. When would someone want to pass in other credentials to method instead of creating another `Transloadit` instance?

Reply:
Removed this now with the exception of the reference implementation